polaris-migrator/src/migrations/styles-insert-stylelint-disable/styles-insert-stylelint-disable.ts

```diff
@@ -0,0 +1,138 @@
+import type {FileInfo} from 'jscodeshift';
+import postcss, {Node, Comment, Plugin, Root} from 'postcss';
+import stylelint, {StylelintPostcssResult} from 'stylelint';
+
+const crossPlatformNewlineRegExp = /\r\n?|\n/;
+const polarisContextMsg = '-- generated by polaris-migrator DO NOT COPY';
+
+const plugin = (): Plugin => {
+  return {
+    postcssPlugin: 'insert-stylelint-disable',
+    Once(root, {result}) {
+      result.messages.forEach(({node, line, endLine}) => {
+        // If a polaris ignore comment exists above the node already,
+        // do nothing
+        if (node.prev()?.text?.includes(polarisContextMsg)) {
+          return;
+        }
+
+        const isMultiline =
+          crossPlatformNewlineRegExp.test(node.value) ||
+          crossPlatformNewlineRegExp.test(node.text) ||
+          line < endLine;
+
+        const commentText = `${
+          isMultiline ? 'stylelint-disable' : 'stylelint-disable-next-line'
+        } ${polarisContextMsg}`;
+
+        const comment = createDisableComment(commentText, node.prev());
+
+        node.before(comment);
+
+        if (isMultiline) {
+          node.type === 'atrule'
+            ? node.prepend(createCommentNode('stylelint-enable'))
+            : node.after(createCommentNode('stylelint-enable'));
+        }
+
+        deleteExtraNewlinesBeforeNode(node);
+      });
+
+      fillDescriptionlessDisables(result, root);
+    },
+  };
+};
+
+export default async function insertStylelintDisable(file: FileInfo) {
+  return postcss([
+    stylelint({
+      config: {
+        extends: ['@shopify/stylelint-polaris'],
+        reportDescriptionlessDisables: true,
```

Comment:
I'm forcing this for now so that I could write a test, but once [stylelint-polaris v5](https://github.com/Shopify/polaris/pull/7691/files) ships we can remove this since it will be part of the `@shopify/stylelint-polaris` config